src/swish/pregexp.ss

```diff
@@ -62,713 +73,684 @@
-                                        (lambda (i1)
-                                          (if (and could-loop-infinitely?
-                                                   (= i1 i))
-                                              (pregexp-error
-                                               'pregexp-match-positions-aux
-                                               'greedy-quantifier-operand-could-be-empty))
-                                          (or (loup-q (+ k 1) i1)
-                                              (fk)))
-                                        fk)
-                                      (or (fk)
-                                          (sub re i
-                                            (lambda (i1)
-                                              (loup-q (+ k 1) i1))
-                                            fk)))))))))))
-               (else (pregexp-error 'pregexp-match-positions-aux))))
-            ((>= i n) (fk))
-            (else (pregexp-error 'pregexp-match-positions-aux))))
-        ;;(printf "done\n")
-        (let ((backrefs (map cdr backrefs)))
-          (and (car backrefs) backrefs)))))
-
-  (define pregexp-replace-aux
-    (lambda (str ins n backrefs)
-      (let loop ((i 0) (r ""))
-        (if (>= i n) r
-            (let ((c (string-ref ins i)))
-              (if (char=? c #\\ )
-                  (let* ((br-i (pregexp-read-escaped-number ins i n))
-                         (br (if br-i (car br-i)
-                                 (if (char=? (string-ref ins (+ i 1)) #\&) 0
-                                     #f)))
-                         (i (if br-i (cadr br-i)
-                                (if br (+ i 2)
-                                    (+ i 1)))))
-                    (if (not br)
-                        (let ((c2 (string-ref ins i)))
-                          (loop (+ i 1)
-                            (if (char=? c2 #\$) r
-                                (string-append r (string c2)))))
-                        (loop i
-                          (let ((backref (pregexp-list-ref backrefs br)))
-                            (if backref
-                                (string-append r
-                                  (substring str (car backref) (cdr backref)))
-                                r)))))
-                  (loop (+ i 1) (string-append r (string c)))))))))
-
-  (define pregexp
-    (lambda (s)
-      (set! *pregexp-space-sensitive?* #t) ;;in case it got corrupted
-      (list ':sub (car (pregexp-read-pattern s 0 (string-length s))))))
+                      identity (lambda () #f))])
+               (if found-it? (sk i) (fk)))]
+            [(:neg-lookahead)
+             (let ([found-it?
+                    (sub (cadr re) i
+                      identity (lambda () #f))])
+               (if found-it? (fk) (sk i)))]
+            [(:lookbehind)
+             (let ([found-it?
+                    (fluid-let ((n i))
```

Comment:
It looks like your new code in :lookbehind and :neg-lookbehind missed the benefits of square brackets.

Reply:
Thanks for finding this!